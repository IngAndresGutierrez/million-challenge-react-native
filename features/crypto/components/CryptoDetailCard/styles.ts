import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  card: {
    borderRadius: 12,
    backgroundColor: '#fff',
  },
  title: {
    marginBottom: 8,
  },
  price: {
    marginBottom: 16,
  },
  divider: {
    marginVertical: 12,
  },
})
