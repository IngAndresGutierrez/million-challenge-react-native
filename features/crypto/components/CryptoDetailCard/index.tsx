import React from 'react'
import { Card, Text, Divider, useTheme } from 'react-native-paper'

import { styles } from './styles'
import { CryptoDetailCardProps } from '../../types'

const CryptoDetailCard = ({ coin }: CryptoDetailCardProps) => {
  const theme = useTheme()

  return (
    <Card style={styles.card} mode="elevated">
      <Card.Content>
        <Text variant="headlineMedium" style={styles.title}>
          {coin.name} ({coin.symbol})
        </Text>

        <Text
          variant="titleLarge"
          style={[styles.price, { color: theme.colors.primary }]}
        >
          USD ${parseFloat(coin.price_usd).toFixed(2)}
        </Text>

        <Divider style={styles.divider} />

        <Text variant="labelLarge">ðŸ“ˆ Price & Ranking</Text>
        <Text>Rank: {coin.rank}</Text>
        <Text>Price (BTC): {coin.price_btc}</Text>

        <Divider style={styles.divider} />

        <Text variant="labelLarge">ðŸ’° Market & Volume</Text>
        <Text>
          Market Cap: ${parseFloat(coin.market_cap_usd).toLocaleString('es-ES')}
        </Text>
        <Text>
          Volume 24h: ${parseFloat(coin.volume24).toLocaleString('es-ES')}
        </Text>
        <Text>
          Volume 24h (prev): $
          {parseFloat(coin.volume24a.toString()).toLocaleString('es-ES')}
        </Text>

        <Divider style={styles.divider} />

        <Text variant="labelLarge">ðŸ“Š Performance</Text>
        <Text>Change 1h: {coin.percent_change_1h}%</Text>
        <Text>Change 24h: {coin.percent_change_24h}%</Text>
        <Text>Change 7d: {coin.percent_change_7d}%</Text>

        <Divider style={styles.divider} />

        <Text variant="labelLarge">ðŸ”— Supply</Text>
        <Text>
          Circulating Supply: {parseFloat(coin.csupply).toLocaleString()}
        </Text>
        <Text>Total Supply: {coin.tsupply || 'N/A'}</Text>
        <Text>Max Supply: {coin.msupply || 'N/A'}</Text>

        <Divider style={styles.divider} />

        <Text variant="labelLarge">ðŸ†” Identifiers</Text>
        <Text>ID: {coin.id}</Text>
        <Text>NameID: {coin.nameid}</Text>
      </Card.Content>
    </Card>
  )
}

export default CryptoDetailCard
