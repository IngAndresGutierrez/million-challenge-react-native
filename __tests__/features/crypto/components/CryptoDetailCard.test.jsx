import React from 'react'
import { render } from '@testing-library/react-native'
import { Provider as PaperProvider } from 'react-native-paper'
import CryptoDetailCard from '../../../../features/crypto/components/CryptoDetailCard'

const mockCoin = {
  id: '1',
  name: 'Bitcoin',
  symbol: 'BTC',
  price_usd: '45000',
  rank: '1',
  market_cap_usd: '850000000000',
  csupply: '19000000',
  volume24: '35000000000',
  percent_change_24h: '2.5',
  nameid: 'bitcoin',
  percent_change_1h: '0.5',
  percent_change_7d: '5.0',
  volume24a: 30000000000,
  tsupply: '21000000',
  msupply: '21000000',
  price_btc: '1.0',
}

describe('CryptoDetailCard', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <PaperProvider>
        <CryptoDetailCard coin={mockCoin} onPress={() => {}} />
      </PaperProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
