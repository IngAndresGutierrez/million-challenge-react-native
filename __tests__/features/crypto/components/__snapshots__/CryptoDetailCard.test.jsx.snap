// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoDetailCard renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3,
        }
      }
      testID="card-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 12,
            "elevation": 1,
            "flex": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          }
        }
        testID="card-container"
      >
        <View
          style={
            [
              {
                "flexShrink": 1,
              },
              undefined,
            ]
          }
          testID="card"
        >
          <View
            borderRadiusStyles={
              {
                "borderRadius": 12,
              }
            }
            style={
              [
                {
                  "paddingHorizontal": 16,
                },
                {
                  "paddingVertical": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 28,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 36,
                    },
                    {
                      "marginBottom": 8,
                    },
                  ],
                ]
              }
            >
              Bitcoin
               (
              BTC
              )
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 22,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 28,
                    },
                    [
                      {
                        "marginBottom": 16,
                      },
                      {
                        "color": "rgba(103, 80, 164, 1)",
                      },
                    ],
                  ],
                ]
              }
            >
              USD $
              45000.00
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  {
                    "marginVertical": 12,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              ðŸ“ˆ Price & Ranking
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Rank: 
              1
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Price (BTC): 
              1.0
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  {
                    "marginVertical": 12,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              ðŸ’° Market & Volume
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Market Cap: $
              850.000.000.000
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Volume 24h: $
              35.000.000.000
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Volume 24h (prev): $
              30.000.000.000
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  {
                    "marginVertical": 12,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              ðŸ“Š Performance
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Change 1h: 
              0.5
              %
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Change 24h: 
              2.5
              %
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Change 7d: 
              5.0
              %
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  {
                    "marginVertical": 12,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              ðŸ”— Supply
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Circulating Supply: 
              19,000,000
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Total Supply: 
              21000000
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              Max Supply: 
              21000000
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  {
                    "marginVertical": 12,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              ðŸ†” Identifiers
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              ID: 
              1
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  undefined,
                ]
              }
            >
              NameID: 
              bitcoin
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
